src/Parser/RetributionPaladin/Modules/Items/Tier21_4set.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatNumber } from 'common/format';
+
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+class Tier21_4set extends Module {
+	static dependencies ={
+		combatants: Combatants,
+	};
+
+	holyPowerGained = 0;
+
+	on_initialized(){
+		this.active = this.combatants.selected.hasBuff(SPELLS.RET_PALADIN_T21_4SET_BONUS.id);
+	}
+
+	on_byPlayer_removebuff(event){
+		const spellId = event.ability.guid;
+		if(spellId !== SPELLS.RET_PALADIN_T21_4SET_BONUS_BUFF.id){
+			return;
+		}
+		this.holyPowerGained += 1;
```

Comment:
Just to confirm the interaction here, the HP Gain is triggered on the debuff being removed.

Reply:
Hmm not technically and I just thought about some cases where this method wouldn't work. I'll fix it

Reply:
I was just trying to understand the interaction. I know nothing about Ret, so if there aren't edge cases where you never drop this buff this should be fine.

Reply:
http://www.wowhead.com/spell=251868/item-paladin-t21-retribution-4p-bonus-judgment-reduces-spender-cost

Actually I was wrong and this method works. Since there is no duration on this buff every time the buff is removed you have cast a HP spender and net 1 holy power. At least according to the information on wowhead.

Reply:
If you think it would make more sense I can check for the cast of a HP spender and check for the buff and then increment the hp gained